src/anshuKJun20/ExtractItemIdFromString.java

```diff
@@ -0,0 +1,61 @@
+package anshuKJun20;
+
+public class ExtractItemIdFromString {
+
+	void findItemId(String input) {
+		// first Apprach;
+		/*
+		 * String replaceStr = input.replace("/", "&"); 
+		 * String[] array = replaceStr.split("&");
+		 *  int num = 0; 
+		 *  for (int index = 0; index < array.length; index++) {
+		 * 	try { 
+		 * 	if (array[index].contains("itemid=12345")) {
+		 *  String[]  temparray = array[index].split("=");
+		 * for (int innerindex = 0; innerindex < temparray.length; innerindex++) {
+		 *   	try { 
+		 *  		num += Integer.parseInt(temparray[innerindex]);
+		 * 		} catch (NumberFormatException ne) {
+		 *   } 
+		 * } 
+		 * }
+		 *  } catch (Exception e) { } } System.out.println(" Item id is " + num);
+		 **/
+
+		// second Approach
+		String number = "";
+		int count = 0;
+		if (input.contains("itemid=")) {
+			String substr = input.substring(input.lastIndexOf("itemid="), input.lastIndexOf("/"));
+			if (substr.contains("empid")) {
+				substr = substr.substring(0, substr.lastIndexOf("&"));
+			}
```

Comment:
if (input.contains("itemId=")) {
			String str = input.split("itemId=")[1];
			String output = "";
			for (int index = 0; index < str.length(); index++) {
				if (Character.isDigit(str.charAt(index))) {
					output += str.charAt(index);
				} else {
					break;   // with this code you don't need to check for empID. 
				}
			}
			System.out.println("Item id is " + output);
		} else {
			System.out.println("No itemID present.");
		}

This code is more optimized.